modules/dynamic-apiserver/pkg/app/app.go

```diff
@@ -175,56 +178,180 @@ func (srv *Service) WaitForServiceBackup(ctx context.Context, serviceId *string,
 	return errors.New("Retries exceeded, backup is not ready")
 }
 
+func (srv *Service) WaitForService(ctx context.Context, serviceId *string, maxRetries int) error {
```

Comment:
I think it would be much better to use WaitForServiceCondition that receives the condition name

Reply:
Removed the function. It was unused.